pinot-core/src/main/java/org/apache/pinot/core/operator/blocks/results/SelectionResultsBlock.java

```diff
@@ -45,6 +56,18 @@ public Collection<Object[]> getRows() {
     return _rows;
   }
 
+  public PriorityQueue<Object[]> getRowsAsPriorityQueue() {
```

Comment:
With the initial results block, we shouldn't need to use this API

Reply:
I find easier to read and to explore the code if we have something like this instead of forcing castings around the code.